### PR TITLE
AO3-6170 Upgrade rails (and actionpack) to 5.2.6

### PR DESCRIPTION
**Pull request by tlee911**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6170

## Purpose

Upgrades rails (and actionpack) to 5.2.6 solving our security issues.

## Testing Instructions

Run all automated tests, verify forms can be submitted.

If you want to be thorough:

- Have a form open with the current rails version before any upgrade is performed (e.g post new work)
- Upgrade the server to the new rails version
- Verify the form can be submitted without issue

Also:
- Have a form open with the new rails version (e.g post new work)
- Downgrade the server back to the previous version
- Verify the form can be submitted without issue

Both of these are possible scenarios in production during the upgrade rollout.
Note that I tried both locally in development and never encountered any issue with either.

## References

https://github.com/rails/rails/blob/5-2-stable/actionpack/CHANGELOG.md

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ gem 'test-unit', '~> 3.2'
 
 gem 'bundler'
 
-gem "rails", "~> 5.2.5"
+gem "rails", "~> 5.2.6"
```
Comment (redsummernight) [inline Gemfile:9]: Can you change this to `~> 5.2`? That way Dependabot can automatically bump patch versions later on.

**Review by redsummernight — COMMENTED** [review]
We are already on 5.2.5, which means we're currently using urlsafe-encoded CSRF tokens. Let's stay with that behavior, as noted in [5.2.6's changelog](https://github.com/rails/rails/releases/tag/v5.2.6). Can you set this config option?

```rb
Rails.application.config.action_controller.urlsafe_csrf_tokens = true
```

**Comment by tlee911** [comment]
> We are already on 5.2.5, which means we're currently using urlsafe-encoded CSRF tokens. Let's stay with that behavior, as noted in [5.2.6's changelog](https://github.com/rails/rails/releases/tag/v5.2.6). Can you set this config option?
> 
> ```ruby
> Rails.application.config.action_controller.urlsafe_csrf_tokens = true
> ```

I'm not sure what's the right place to put this

1. In every environment config file (e.g. production.rb, staging.rb, etc.)?
2. In environment.rb itself?
3. In a new initializer file (none of the existing ones seem to fit)?
4. Some other place I haven't considered?

**Comment by redsummernight** [comment]
Let's set it in config/application.rb.

**Review by redsummernight — APPROVED** [review]
Thanks!

**Pull request merged**